### PR TITLE
Rework provisioners vars to support different prefix/version for Origin/OSE

### PR DESCRIPTION
**Pull request by vrutkovs**
Fixes bug https://bugzilla.redhat.com/show_bug.cgi?id=1523534
Obsoletes https://github.com/openshift/openshift-ansible/pull/6396

### CODE REVIEW

File: roles/openshift_provisioners/tasks/main.yaml [hunk 1]
```diff
@@ -12,6 +12,19 @@
...
+- name: Set default image variables based on openshift_deployment_type
+  include_vars: "{{ var_file_name }}"
+  with_first_found:
+    - "{{ openshift_deployment_type }}.yml"
+    - "default_images.yml"
+  loop_control:
+    loop_var: var_file_name
+
+- name: Set eventrouter image facts
+  set_fact:
```
Comment (michaelgugino) [inline roles/openshift_provisioners/tasks/main.yaml]: I prefer not to use set_fact unless it's 100% necessary.  This means that some value we dynamically derive is need across play scopes.

I prefer to use dictionaries for this type of small thing:

https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_facts/defaults/main.yml#L4-L15

File: roles/openshift_provisioners/defaults/openshift-enterprise.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+__openshift_provisioners_image_prefix: "registry.access.redhat.com/openshift3/ose-"
```
Comment (michaelgugino) [inline roles/openshift_provisioners/defaults/openshift-enterprise.yml]: I'm not a fan of the double underscore variables.  We were using l_x (lowercase L) prefix for variables we don't want users to muck with.  I realize some roles do that, but it's not my preference.

Comment (vrutkovs) [reply roles/openshift_provisioners/defaults/openshift-enterprise.yml]: Fixed, using dicts now

**Pull request merged**